linera-service/src/linera-exporter/storage.rs

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{error::Error, sync::Arc};
+
+use linera_base::{crypto::CryptoHash, data_types::Blob, identifiers::BlobId};
+use linera_chain::types::ConfirmedBlock;
+use linera_sdk::{
+    bcs,
+    views::{View, ViewError},
+};
+use linera_service::storage::StoreConfig;
+use linera_storage::BaseKey;
+use linera_storage_service::client::ServiceStoreClient;
+#[cfg(feature = "dynamodb")]
+use linera_views::dynamo_db::DynamoDbStore;
+#[cfg(feature = "scylladb")]
+use linera_views::scylla_db::ScyllaDbStore;
+use linera_views::{
+    batch::Batch,
+    context::ViewContext,
+    memory::{MemoryStore, MemoryStoreConfig},
+    rocks_db::RocksDbStore,
+    store::KeyValueStore,
+};
+#[cfg(with_testing)]
+use linera_views::{random::generate_test_namespace, store::TestKeyValueStore};
+#[cfg(all(feature = "rocksdb", feature = "scylladb"))]
+use {linera_storage::ChainStatesFirstAssignment, linera_views::backends::dual::DualStore};
+
+use crate::{state::BlockExporterStateView, ExporterContext, ExporterError, Generic};
+
+const BASE_KEY: &str = "exporter";
+
+/// Extension for the [`DbStorage`] type to abstract the persistant storage and operate in the context of [`BlockExporterStateView`].
+#[derive(Clone)]
+pub(crate) struct ExporterStorage<Store> {
+    store: Arc<Store>,
+}
+
+impl<Store> ExporterStorage<Store>
+where
+    Store: KeyValueStore + Clone + Send + Sync + 'static,
+    Store::Error: Error + Send + Sync,
+{
+    pub(super) async fn load_exporter_state(
+        &self,
+    ) -> Result<BlockExporterStateView<ViewContext<(), Store>>, ViewError> {
+        let root_key = bcs::to_bytes(BASE_KEY)?;
+        let store = self.store.clone_with_root_key(&root_key)?;
+        let context = ViewContext::create_root_context(store, ()).await?;
+        BlockExporterStateView::load(context).await
+    }
+
+    pub(super) async fn contains_blob(&self, blob_id: BlobId) -> Result<bool, ViewError> {
```

Comment:
I see your problem and created https://github.com/linera-io/linera-protocol/pull/3748 to avoid duplicating all this code here.